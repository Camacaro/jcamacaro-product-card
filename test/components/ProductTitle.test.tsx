import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src'
import { product1 } from '../data/products';

describe('ProdcutTitle', () => {
  it('debe mostrar el componente correctamente con el titulo personalizado', () => {
    const wrapper = renderer.create(
      <ProductTitle title="Custom Producto" />
    )

    // console.log(wrapper.toJSON())
    expect( wrapper.toJSON() ).toMatchSnapshot()
  });

  it('debe de mostrar el componente con el nombre del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {() => (
          <ProductTitle />
        )}
      </ProductCard>
    )

    // console.log(wrapper.toJSON())
    expect( wrapper.toJSON() ).toMatchSnapshot()
  });
});
