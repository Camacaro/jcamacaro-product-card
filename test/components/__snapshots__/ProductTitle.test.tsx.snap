// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProdcutTitle debe de mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Card
  </span>
</div>
`;

exports[`ProdcutTitle debe mostrar el componente correctamente con el titulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Custom Producto
</span>
`;
